src/main/java/uk/gov/laa/ccms/caab/service/ApplicationService.java

```diff
@@ -534,55 +544,204 @@ public ResultsDisplay<LinkedCaseResultRowDisplay> getLinkedCases(final String ap
+
+    for (final uk.gov.laa.ccms.caab.model.ScopeLimitation scopeLimitation : scopeLimitations) {
+      final ScopeLimitationDetail criteria = new ScopeLimitationDetail()
+          .categoryOfLaw(categoryOfLaw)
+          .matterType(matterType)
+          .proceedingCode(proceedingCode)
+          .levelOfService(levelOfService)
+          .scopeLimitations(scopeLimitation.getScopeLimitation().getId());
+
+      if (EMERGENCY_APPLICATION_TYPE_CODES.contains(applicationType)) {
+        criteria.emergency(true);
+      }
+
+      lookupService.getScopeLimitationDetails(criteria)
+          .block()
+          .getContent()
+          .stream()
+          .findFirst()
+          .map(ScopeLimitationDetail::getCostLimitation)
+          .ifPresent(costLimitation -> costLimitations.add(costLimitation.floatValue()));
+
+    }
+
+    if (!costLimitations.isEmpty()) {
+      maxValue = BigDecimal.valueOf(
+          costLimitations
+              .stream()
+              .max(Float::compareTo)
+              .orElse(null));
+    }
+
+    return maxValue;
   }
 
   /**
-   * Fetches the prior authorities associated with a specific application id.
-   * This method communicates with the CAAB API client to fetch the prior authorities and
-   * transforms them into a {@code ResultsDisplay<PriorAuthority>} format.
+   * Determines the proceeding stage based on various parameters.
    *
-   * @param applicationId The id of the application for which prior authorities should be retrieved.
-   * @return A {@code Mono<ResultsDisplay<PriorAuthority>>} containing the prior authorities.
+   * @param categoryOfLaw The category of law.
+   * @param matterType The type of the matter.
+   * @param proceedingCode The code of the proceeding.
+   * @param levelOfService The level of service.
+   * @param scopeLimitations The list of scope limitations.
+   * @param isAmendment Flag indicating if it's an amendment.
+   * @return The proceeding stage as an Integer.
    */
-  public Mono<ResultsDisplay<PriorAuthority>> getPriorAuthorities(final String applicationId) {
-    final ResultsDisplay<PriorAuthority> results = new ResultsDisplay<>();
+  public Integer getProceedingStage(
+      final String categoryOfLaw,
+      final String matterType,
+      final String proceedingCode,
+      final String levelOfService,
+      final List<uk.gov.laa.ccms.caab.model.ScopeLimitation> scopeLimitations,
+      final boolean isAmendment) {
+
+    // String existingStage = null;
+    //todo - see GetDefaultScopeLimitation in pui
+    //    if (isAmendment) {
+    //
+    //      // for (Proceeding caseProceeding : myCase.getProceedings()) {
+    //      //   if (caseProceeding.getEbsId().equalsIgnoreCase(proceeding.getEbsId())) {
+    //      //     existingStage = caseProceeding.getStage();
+    //      //   }
+    //      // }
+    //      // see - getAmendmentProceedingStage
+    //    }
+
+    if (scopeLimitations.size() == 1) {
```

Comment:
Can this block not be removed, and the following bit is only done once if there's only one scopeLimitation?

Reply:
removed if